uaa/client.go

```diff
@@ -95,6 +95,10 @@ func (c *Client) CreateClient(clientID, name, spaceGUID string) (map[string]stri
 		// UAA does not allow `implicit` clients to be created without a redirect uri
 		m["redirect_uri"] = placeholderRedirectURI
 	} else {
+		if strings.Contains(grantTypes, "authorization_code") {
```

Comment:
This can be rolled up to the above check and be updated to `if strings.Contains(grantTypes, "implicit") || strings.Contains(grantTypes, "authorization_code")`

Reply:
It could be but we have to be sure that in case of `authorization_code` we generate a client_secret. That is why I duplicated it. I will redo it now .